scripts/tests-after-upgrade/tests-after-upgrade.sh

```diff
@@ -0,0 +1,295 @@
+    check_response "$result" "\"subject\": \"$noc_root_cert_subject\""
+    check_response "$result" "\"subjectKeyId\": \"$noc_root_cert_subject_key_id\""
+
+  fi
+
+  echo "Get certificates (DA)"
+  result=$(dcld query pki all-x509-certs)
+  check_response "$result" "\"subject\": \"$da_root_cert_subject\""
+
+  echo "Get certificate (DA)"
+  result=$(dcld query pki x509-cert --subject="$da_root_cert_subject" --subject-key-id="$da_root_cert_subject_key_id")
+  check_response "$result" "\"subject\": \"$da_root_cert_subject\""
+  check_response "$result" "\"subjectKeyId\": \"$da_root_cert_subject_key_id\""
+
+  test_divider
+
+  # Validator Node
+
+  echo "Get validator"
+  result=$(dcld query validator node --address="$validator_address")
+  check_response "$result" "\"owner\": \"$validator_address\""
+
+  echo "Get all validators"
+  result=$(dcld query validator all-nodes)
+  check_response "$result" "\"owner\": \"$validator_address\""
+
+  test_divider
+}
+
+test_write_requests() {
+
+  # Write commands for vendor test account
+
+  if [ "$ENVIRONMENT" = "local" ]; then
+    :
+    # create_new_vendor_account $vendor_account $vid_vendor
+  fi
+
+  if [ "$ENVIRONMENT" != "mainnet" ]; then
+
+    echo "Use keys for a $vendor_account"
+    result="$(echo "$passphrase" | dcld keys add "$vendor_account" --recover)"
+
+    # MODEL and MODEL_VERSION
+
+    pid_random=$(random_four_digit_int)
+    echo "Add model vid=$vid_vendor pid=$pid_random"
+    result=$(dcld tx model add-model --vid="$vid_vendor" --pid="$pid_random" --deviceTypeID="$device_type_id" --productName="$product_name" --from="$vendor_account" --yes)
+    result=$(get_txn_result "$result")
+    check_response "$result" "\"code\": 0"
+
+    echo "Add model version vid=$vid_vendor pid=$pid_random"
+    result=$(dcld tx model add-model-version --vid="$vid_vendor" --pid="$pid_random" --softwareVersion="$software_version" --softwareVersionString="$software_version_string" --cdVersionNumber="$cd_version_number" --minApplicableSoftwareVersion="$min_applicable_software_version" --maxApplicableSoftwareVersion="$max_applicable_software_version" --from="$vendor_account" --yes)
+    result=$(get_txn_result "$result")
+    check_response "$result" "\"code\": 0"
+
+    # X509 PKI
+
+    echo "Add NOC Root certificate by vendor with VID = $vendor_account"
+    result=$(dcld tx pki add-noc-x509-root-cert --certificate="$noc_root_cert" --from "$vendor_account" --yes)
+    result=$(get_txn_result "$result")
+    check_response "$result" "\"code\": 0"
+
+    test_divider
+
+    cleanup() {
+      echo "Teardown: delete the added NOC Root certificate"
+      result=$(dcld tx pki remove-noc-x509-root-cert --subject="$subject" --subject-key-id="$subject_key_id" --from "$vendor_account" --yes)
+      result=$(get_txn_result "$result")
+      check_response "$result" "\"code\": 0"
+
+      echo "Teardown: delete keys for a $vendor_account"
+      result=$(dcld keys delete "$vendor_account" --yes)
+    }
```

Comment:
We can move the definition of `cleanup` method right after adding the keys
And call it right after with `trap cleanup EXIT`
Then, if something go wrong in the test, we can be sure that keys and certificate will be removed